jobs: TestMetrics failed

jobs.TestMetrics [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3017249&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3017249&tab=artifacts#/) on master @ [821d3153b6925104a692b28ad21bb2e82156f051](https://github.com/cockroachdb/cockroach/commits/821d3153b6925104a692b28ad21bb2e82156f051):


```
=== RUN   TestMetrics
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestMetrics832649711
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestMetrics
    jobs_test.go:2681: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestMetrics832649711
--- FAIL: TestMetrics (2.33s)
=== RUN   TestMetrics/restart,_pause,_resume,_then_success
E210526 18:21:11.442943 81432 jobs/adopt.go:284  [n1] 7  job 661857473033502721: adoption completed with error job 661857473033502721: : restarting in background
    jobs_test.go:2597: 
        	Error Trace:	jobs_test.go:2597
        	Error:      	Received unexpected error:
        	            	job with status pause-requested cannot be resumed
        	Test:       	TestMetrics/restart,_pause,_resume,_then_success
E210526 18:21:11.511114 80515 jobs/registry.go:697  [-] 8  error claiming jobs: could not query jobs table: claim-jobs: context canceled
E210526 18:21:11.511417 80515 jobs/registry.go:707  [-] 9  error processing claimed jobs: could not query for claimed jobs: select-running/get-claimed-jobs: context canceled
E210526 18:21:11.548571 80433 jobs/registry.go:687  [-] 10  failed to serve pause and cancel requests: node unavailable; try another peer
    --- FAIL: TestMetrics/restart,_pause,_resume,_then_success (0.52s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestMetrics PKG=./pkg/jobs TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-json
</p>
</p>
</details>
/cc @cockroachdb/sql-schema ajwerner
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMetrics.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/jobs/jobs_test.go
+++ b/pkg/jobs/jobs_test.go
@@ -2534,18 +2534,20 @@ func TestMetrics(t *testing.T) {
 	jobs.RegisterConstructor(jobspb.TypeImport, func(_ *jobs.Job, _ *cluster.Settings) jobs.Resumer {
 		return res
 	})
-	setup := func(t *testing.T) (s serverutils.TestServerInterface, r *jobs.Registry, cleanup func()) {
+	setup := func(t *testing.T) (
+		s serverutils.TestServerInterface, db *gosql.DB, r *jobs.Registry, cleanup func(),
+	) {
 		jobConstructorCleanup := jobs.ResetConstructors()
-		s, _, _ = serverutils.StartServer(t, base.TestServerArgs{})
+		s, db, _ = serverutils.StartServer(t, base.TestServerArgs{})
 		r = s.JobRegistry().(*jobs.Registry)
-		return s, r, func() {
+		return s, db, r, func() {
 			jobConstructorCleanup()
 			s.Stopper().Stop(ctx)
 		}
 	}
 
 	t.Run("success", func(t *testing.T) {
-		_, registry, cleanup := setup(t)
+		_, _, registry, cleanup := setup(t)
 		defer cleanup()
 		rec := jobs.Record{
 			DescriptorIDs: []descpb.ID{1},
@@ -2561,7 +2563,7 @@ func TestMetrics(t *testing.T) {
 		int64EqSoon(t, backupMetrics.ResumeCompleted.Count, 1)
 	})
 	t.Run("restart, pause, resume, then success", func(t *testing.T) {
-		_, registry, cleanup := setup(t)
+		_, db, registry, cleanup := setup(t)
 		defer cleanup()
 		rec := jobs.Record{
 			DescriptorIDs: []descpb.ID{1},
@@ -2593,6 +2595,12 @@ func TestMetrics(t *testing.T) {
 			require.Equal(t, int64(0), importMetrics.CurrentlyRunning.Value())
 		}
 		{
+			// Wait for the job to be marked paused.
+			tdb := sqlutils.MakeSQLRunner(db)
+			q := fmt.Sprintf("SELECT status FROM system.jobs WHERE id = %d", jobID)
+			tdb.CheckQueryResultsRetry(t, q, [][]string{{"paused"}})
+		}
+		{
 			// Now resume the job and let it succeed.
 			require.NoError(t, registry.Unpause(ctx, nil, jobID))
 			errCh := <-resuming
@@ -2602,7 +2610,7 @@ func TestMetrics(t *testing.T) {
 		}
 	})
 	t.Run("failure then restarts in revert", func(t *testing.T) {
-		_, registry, cleanup := setup(t)
+		_, _, registry, cleanup := setup(t)
 		defer cleanup()
 		rec := jobs.Record{
 			DescriptorIDs: []descpb.ID{1},
@@ -2637,7 +2645,7 @@ func TestMetrics(t *testing.T) {
 		}
 	})
 	t.Run("fail, pause, resume, then success on failure", func(t *testing.T) {
-		_, registry, cleanup := setup(t)
+		_, db, registry, cleanup := setup(t)
 		defer cleanup()
 		rec := jobs.Record{
 			DescriptorIDs: []descpb.ID{1},
@@ -2667,6 +2675,12 @@ func TestMetrics(t *testing.T) {
 			require.Equal(t, int64(1), importMetrics.ResumeFailed.Count())
 			require.Equal(t, int64(0), importMetrics.ResumeCompleted.Count())
 			require.Equal(t, int64(0), importMetrics.CurrentlyRunning.Value())
+		}
+		{
+			// Wait for the job to be marked paused.
+			tdb := sqlutils.MakeSQLRunner(db)
+			q := fmt.Sprintf("SELECT status FROM system.jobs WHERE id = %d", jobID)
+			tdb.CheckQueryResultsRetry(t, q, [][]string{{"paused"}})
 		}
 		{
 			// Now resume the job and let it succeed.

